src/haddock/modules/topology/topocg/cns/build-missing.cns

```diff
@@ -0,0 +1,296 @@
+        end
+
+        {- turn on initial energy terms -}
+        flags exclude * include bond angle vdw end
+ 
+        minimize powell nstep=50  nprint=10 end
+
+        set seed=$iniseed end
+
+        do (vx=maxwell($bath)) (store9)
+        do (vy=maxwell($bath)) (store9)
+        do (vz=maxwell($bath)) (store9)
+
+        flags exclude vdw include impr end
+
+        dynamics cartesian
+            nstep=50
+            timestep=$timestep
+            tcoupling=true temperature=$bath
+            nprint=$nstep
+            cmremove=false
+        end
+
+        flags include vdw end
+
+        minimize powell nstep=50 nprint=10 end
+
+        set seed=$iniseed end
+
+        do (vx=maxwell($bath)) (store9)
+        do (vy=maxwell($bath)) (store9)
+        do (vz=maxwell($bath)) (store9)
+
+        dynamics cartesian
+            nstep=50
+            timestep=$timestep
+            tcoupling=true temperature=$bath
+            nprint=$nstep
+            cmremove=false
+        end
+
+        parameter
+            nbonds
+                rcon=2. nbxmod=-3 repel=0.75
+            end
+        end
+
+        minimize powell nstep=100 nprint=25 end
+
+        set seed=$iniseed end
+
+        do (vx=maxwell($bath)) (store9)
+        do (vy=maxwell($bath)) (store9)
+        do (vz=maxwell($bath)) (store9)
+
+        dynamics cartesian
+            nstep=$nstep
+            timestep=$timestep
+            tcoupling=true temperature=$bath
+            nprint=$nstep
+            cmremove=false
+        end
+
+        {- turn on all energy terms -}
+        flags include dihe ? end
+
+        {- set repel to ~vdw radii -}
+        parameter
+            nbonds
+                repel=0.89
+            end
+        end
+
+        minimize powell nstep=500 nprint=50 end
+
+        flags exclude * include bond angl impr dihe vdw elec end
+
+        minimize powell nstep=500 nprint=50 end
+
+        {- return masses to something sensible -}
+        do (mass=refy) (store9)
+
+        set seed=$iniseed end
+
+        do (vx=maxwell($bath)) (store9)
+        do (vy=maxwell($bath)) (store9)
+        do (vz=maxwell($bath)) (store9)
+
+        dynamics cartesian
+            nstep=$nstep
+            timestep=$timestep
+            tcoupling=true temperature=$bath
+            nprint=$nstep
+            cmremove=false
+        end
+
+        {- some final minimisation -}
+        minimize powell nstep=500 drop=40.0 nprint=50 end
+
+        print thres=0.02 bonds
+        print thres=5. angles
+
+    end if
+  
+    noe reset end
+    rest dihe reset end
+
+
+    ! Check for D-amino acids
+
+    for $id in id ( name HA and not resn GLY ) loop ltod
+
+        show (segid) (id $id)
+        evaluate ($segid=$result)
+        show (resid) (id $id)
+        evaluate ($resid=$result)
+        show (resn) (id $id)
+        evaluate ($resn=$result)
+
+        show ave(x) (segid $segid and resid $resid and (name CB or name C or name N) and known)
```

Comment:
Non existing atom names in CG mode. May-be irrelevant if D-amino acids are not supported

Reply:
Can be removed